src/WebJobs.Extensions.Http/ClaimsPrincipalBindingProvider.cs

```diff
@@ -0,0 +1,108 @@
+namespace Microsoft.Azure.WebJobs.Extensions.Http
+{
+    /// <summary>
+    /// This provider provides a binding to Type <see cref="ClaimsPrincipal"/>.
+    /// </summary>
+    internal class ClaimsPrincipalBindingProvider : IBindingProvider
+    {
+        public Task<IBinding> TryCreateAsync(BindingProviderContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            if (context.Parameter.ParameterType != typeof(ClaimsPrincipal))
+            {
+                return Task.FromResult<IBinding>(null);
+            }
+
+            return Task.FromResult<IBinding>(new ClaimsIdentityBinding(context.Parameter));
+        }
+
+        private class ClaimsIdentityBinding : IBinding
+        {
+            private readonly ParameterInfo _parameter;
+
+            public ClaimsIdentityBinding(ParameterInfo parameter)
+            {
+                _parameter = parameter;
+            }
+
+            public bool FromAttribute
+            {
+                get { return false; }
+            }
+
+            public Task<IValueProvider> BindAsync(BindingContext context)
+            {
+                if (context == null)
+                {
+                    throw new ArgumentNullException("context");
+                }
+
+                return BindInternalAsync();
+            }
+
+            public Task<IValueProvider> BindAsync(object value, ValueBindingContext context)
+            {
+                if (context == null)
+                {
+                    throw new ArgumentNullException("context");
+                }
+
+                return BindInternalAsync();
+            }
+            
+            private static Task<IValueProvider> BindInternalAsync()
+            {
+                return Task.FromResult<IValueProvider>(new ClaimsPrincipalValueProvider());
+            }
+
+            public ParameterDescriptor ToParameterDescriptor()
+            {
+                return new ParameterDescriptor
+                {
+                    Name = _parameter.Name,
+                    DisplayHints = new ParameterDisplayHints
+                    {
+                        Description = "ClaimsPrincipal"
+                    }
+                };
+            }
+
+            private class ClaimsPrincipalValueProvider : IValueProvider
+            {
+                public ClaimsPrincipalValueProvider()
+                {
+                }
+
+                public Type Type
+                {
+                    get { return typeof(ClaimsPrincipal); }
+                }
+
+                public Task<object> GetValueAsync()
+                {
+                    return Task.FromResult<object>(ClaimsPrincipal.Current);
+                }
+
+                public string ToInvokeString()
+                {
+                    // TODO: figure out right value here
+                    return ClaimsPrincipal.Current.ToString();
```

Comment:
Probably `ClaimsPrincipal.Identity.Name` is what you want.

Reply:
Do we need to check for null here?

---
In reply to: [171935462](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/373#discussion_r171935462) [](ancestors = 171935462)